test_fms/fms2_io/test_get_is_valid.F90

```diff
@@ -0,0 +1,97 @@
+!***********************************************************************
+!*                   GNU Lesser General Public License
+!*
+!* This file is part of the GFDL Flexible Modeling System (FMS).
+!*
+!* FMS is free software: you can redistribute it and/or modify it under
+!* the terms of the GNU Lesser General Public License as published by
+!* the Free Software Foundation, either version 3 of the License, or (at
+!* your option) any later version.
+!*
+!* FMS is distributed in the hope that it will be useful, but WITHOUT
+!* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+!* FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+!* for more details.
+!*
+!* You should have received a copy of the GNU Lesser General Public
+!* License along with FMS.  If not, see <http://www.gnu.org/licenses/>.
+!***********************************************************************
+
+!> @brief  This programs tests calls to fms2_io get_valid and is_valid
+
+program test_get_is_valid
+
+use   fms2_io_mod, only: fms2_io_init, open_file, FmsNetcdfFile_t, valid_t, &
+                         close_file, is_valid, get_valid
+use   mpp_mod    , only: mpp_init, mpp_exit, mpp_npes, mpp_get_current_pelist, &
+                         mpp_root_pe, mpp_pe, FATAL, mpp_error
+use   netcdf     , only: nf90_create, nf90_def_var, nf90_put_att, nf90_enddef, &
+                         nf90_close, nf90_clobber, nf90_64bit_offset, nf90_double
+
+use, intrinsic :: iso_fortran_env
+
+implicit none
+
+type(valid_t) :: valid_type               !> Fms2io valid object type
+type(FmsNetcdfFile_t) :: fileobj          !> Fms2io file obj
+real(kind=real64) :: sst(10,10)           !> Data
+integer :: mask_in(10,10)                 !> Array defining if sst(:,:) is valid (1) or not (0)
+integer :: ncid, varid, err               !> Netcdf integers needed
+integer, dimension(:), allocatable :: pes !> Current pelist
+
+call mpp_init()
```

Comment:
This needs an mpp_init  test_level [parameter](https://github.com/NOAA-GFDL/FMS/blob/40c6cb9663f574b902ee8d832828be8f04c64feb/mpp/mpp.F90#L1368). 
You'll have to replace `mpp_exit()` with `MPI_FINALIZE(ierr)`.

Reply:
why? the `test_level` flag was added for when we were testing subroutines used by `mpp_init`, this is not doing that. 

This test is testing `get_valid` and `is_valid`.

Reply:
Ah, my mistake. Ignore this request.